apps/mobile/src/state/persisted.ts

```diff
@@ -0,0 +1,58 @@
+import * as SecureStore from 'expo-secure-store';
+import { StateCreator } from 'zustand';
+import { createJSONStorage, persist } from 'zustand/middleware';
+
+export const KEY_PREFIX = 'PERSISTED_';
+
+interface PersistedState {
+  theme: string;
+  _hasHydrated: boolean;
+}
+
+interface PersistedAction {
+  setTheme(theme: string): void;
+  setHasHydrated(state: boolean): void;
+}
+
+export type PersistedStore = PersistedState & PersistedAction;
+
+export const createPersistedStore: StateCreator<
```

Comment:
Excuse my ignorance but where do we persist this data? To `AsyncStorage`?

I think it may not be a bad idea to just encrypt everything to protected storage